Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -595,4 +595,301 @@ ROMol *fragmentOnBRICSBonds(const ROMol &mol) {
+  case MolzipLabel::FragmentOnBonds:
+      // shouldn't ever get here
+      assert(0);
+      break;
+  default:
+      CHECK_INVARIANT(0,"bogus MolZipLabel value in MolZip::get_label");
+  }
+  }
+  return idx;
+}
+
+// Return the connected atom
+//  n.b. There can be only one connection from a mapped atom
+Atom *get_other_atom(Atom *a) {
+  PRECONDITION(a, "null atom in MolZip::get_other_atom");
+  auto &m = a->getOwningMol();
+  if(m.getAtomDegree(a) != 1)
+    return nullptr;
+  
+  for(const auto &nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(a))) {
+    return m[nbrIdx];
+  }
+  return nullptr; // can't get here
+}
+
+int num_swaps_to_interconvert(std::vector<unsigned int> &orders) {
+    int nswaps = 0;
+    std::vector<bool> seen(orders.size());
+    for(size_t i=0;i<orders.size();++i) {
+        if(!seen[i]) {
+            auto j = i;
+            while(orders[j] != i) {
+                j=orders[j];
+                seen[j] = true;
+                nswaps++;
+            }
+        }
+    }
+    return nswaps;
+}
+
+// Simple bookkeeping class to bond attachments and handle stereo
+struct ZipBond {
+    Atom *a;       // atom being bonded
+    Atom *a_dummy; // Labelled atom, i.e. [*:1]-C  will bond the C to something
+    Atom *b;       // atom being bonded
+    Atom *b_dummy; // Labelled atom, i.e. [*:1]-O will bond the O to something
+    
+    // Mark the original order of the nbr atoms including the dummy
+    //  The goal is to copy the dummy chiral order over to the
+    //  atom being bonded
+    void mark(Atom *chiral_atom, Atom *dummy_atom, Atom *new_atom) const {
+        if(chiral_atom->getChiralTag()) {
+            std::string mark = "__molzip_mark_" + std::to_string(chiral_atom->getIdx());
+            chiral_atom->setProp("__molzip_chiral_mark", mark);
+            int order = 0;
+            auto &m = chiral_atom->getOwningMol();
+            for(auto nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(chiral_atom))) {
+                m[nbrIdx]->setProp<int>(mark, order);
+                ++order;
+            }
+            new_atom->setProp<int>(mark, dummy_atom->getProp<int>(mark));
+        }
+        
+        // check bond stereo
+        auto &m = chiral_atom->getOwningMol();
+        auto bond  = m.getBondBetweenAtoms(chiral_atom->getIdx(), dummy_atom->getIdx());
+        for(auto nbrIdx : boost::make_iterator_range(m.getAtomNeighbors(chiral_atom))) {
+            auto bond  = m.getBondBetweenAtoms(chiral_atom->getIdx(), nbrIdx);
+            if(bond->getStereo()) {
+                std::string mark = "__molzip_bond_stereo_mark";
+                std::vector<Atom*> atoms;
+                bool has_dummy = false;
+                for(auto idx : bond->getStereoAtoms()) {
+                    if(idx == dummy_atom->getIdx()) {
+                        atoms.push_back(new_atom);
+                        has_dummy = true;
+                    } else {
+                        atoms.push_back(m[idx]);
```

Comment:
Though this technical works, it's relying on an implementation detail (that node iterators are equal to atom indices), this would be much safer and more explicit as:
```suggestion
                        atoms.push_back(m.getAtomWithIdx(idx));
```

Reply:
Ah, I see your point.  Took me a bit.